File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int major = 0 , count = 0;

        for (int num : nums){
            if (count == 0){
                count = 1;
                major = num;
            }            
            else if (num == major) count ++;
            else count--;
        }
        return major ;

        // beware
        // int major = nums[0];
        // for(int i = 1; i < nums.length; i++){
        //     int count = 1;
        //     if(nums[i] == major) count++;
        //     else count--;
        //     if(count <= 0) major = nums[i];
        // }
        // return major;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major = nums[0];
        int count = 0;
        for(int i = 1 ; i < nums.size(); i++){
            if(nums[i] == major){
                count++;
            }
            else count--;
            if(count < 0){
                major = nums[i];
                count = 0;
            }
        }
        return major;
    }
};